domain/src/main/java/org/oppia/domain/oppialogger/loguploader/LogUploadWorkRequest.kt

```diff
@@ -0,0 +1,80 @@
+package org.oppia.domain.oppialogger.loguploader
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequest
+import androidx.work.WorkManager
+import org.oppia.domain.oppialogger.ApplicationStartupListener
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val OPPIA_EXCEPTION_WORK = "OPPIA_EXCEPTION_WORK_REQUEST"
+private const val OPPIA_EVENT_WORK = "OPPIA_EVENT_WORK_REQUEST"
+
+/** Enqueues unique periodic work requests for uploading events and exceptions to the remote service on application creation. */
+@Singleton
+class LogUploadWorkRequest @Inject constructor(
+  private val context: Context
+) : ApplicationStartupListener {
+
+  private val workerConstraints = Constraints.Builder()
+    .setRequiredNetworkType(NetworkType.CONNECTED)
+    .setRequiresBatteryNotLow(true)
+    .setRequiresStorageNotLow(true)
```

Comment:
Why this constraint? Won’t this prevent uploading logs in cases when storage is low and could be freed by uploading logs?

Reply:
Okay, done.